CHANGELOG.md

```diff
@@ -26,6 +26,45 @@
 - A warning is now emitted if an object has finalizers which might be blocking
   deletion. (https://github.com/pulumi/pulumi-kubernetes/issues/1418)
 
+- The `pulumi.com/waitFor` annotation was introduced to allow for custom
+  readiness checks. This override Pulumi's own await logic for the resource
+  (however the `pulumi.com/skipAwait` annotation still takes precedence).
+  
+  The value of this annotation can take 3 forms:
+    1. A string prefixed with `jsonpath=` followed by a
+       [JSONPath](https://kubernetes.io/docs/reference/kubectl/jsonpath/)
+       expression and an optional value.
+
+       The JSONPath expression accepts the same syntax as 
+       `kubectl get -o jsonpath={...}`.
+       
+       If a value is provided, the resource is considered ready when the
+       JSONPath expression evaluates to the same value. For example this
+       resource expects its "phase" field to have a value of "Running":
+     
+           `pulumi.com/waitFor: "jsonpath={.phase}=Running"` 
+       
+       If a value is not provided, the resource will be considered ready when
+       any value exists at the given path. This resource will wait until it has
```

Comment:
From the code, it looks like an empty string as a value would also be considered ready since it always returns true if the path exists.

```
// Matches returns true if the JSONPath matches against the given object. If
// the JSONPath didn't include a value, then this will return true if the path
// exists. Otherwise, the path must exist and hold a value equal to the
// Instance's expected value.
```

@blampe Can we add an additional test case to `provider/pkg/await/condition/jsonpath_test.go` to confirm/clarify this behaviour?

Reply:
@rquitales for sure! FWIW this is based on https://github.com/kubernetes/kubectl/blob/c4be63c54b7188502c1a63bb884a0b05fac51ebd/pkg/cmd/wait/json.go#L72-L91 which succeeds if the path exists with any value.